Services/Database/classes/class.ilAuthContainerMDB2.php

```diff
@@ -49,7 +49,7 @@ public function __construct()
 
 		// studip mode: check against submitted md5 password for ilSoapUserAdministration::login()
 		// todo: check whether we should put this to another place
-		if (isset($_POST['password']) && (preg_match('/^[a-f0-9]{32,32}$/i', $_POST['password']) || (strlen($_POST['password']) == 60)))
+		if (isset($_POST['password']) && (preg_match('/^[a-f0-9]{32,32}$/i', $_POST['password']) || preg_match('/\$2[aby]\$\d{2}\$.{53}/i', $_POST['password'])))
 		{
```

Comment:
We could reduce the execution of this code to the soap context:
include_once "Services/Context/classes/class.ilContext.php";
if(ilContext::getType() == ilContext::CONTEXT_SOAP)